src/shared/sol-conffile.c

```diff
@@ -58,93 +61,198 @@ sol_conffile_entry_sort_cb(const void *data1, const void *data2)
     return strcasecmp(e1->id, e2->id);
 }
 
+static char**
+str_split(const char* a_str, const char a_delim)
+{
+    char** result = 0;
+    size_t count = 0;
+    char* tmp_counter = 0;
+    char* tmp = 0;
+    char* last_comma = 0;
+    char delim[2];
+    delim[0] = a_delim;
+    delim[1] = 0;
+
+    tmp = strdup(a_str);
+    tmp_counter = tmp;
+    while (*tmp_counter) {
+        if (a_delim == *tmp_counter) {
+            count++;
+            last_comma = tmp_counter;
+        }
+        tmp_counter++;
+    }
+
+    count += last_comma < (tmp + strlen(tmp) - 1); // space for last comma
+    count += 1; // space for null
+
+    result = malloc(sizeof(char*) * count);
+    if (result) {
+        size_t idx  = 0;
+        char* token = strtok(tmp, delim);
+        while (token) {
+            SOL_INT_CHECK(idx, >= count, NULL);
+            *(result + idx++) = strdup(token);
+            token = strtok(0, delim);
+        }
+        SOL_INT_CHECK(idx, == count, NULL);
+        *(result + idx) = 0;
+    }
+
+    free (tmp);
+    return result;
+}
+
+static char * dup_string_from_json_token(struct sol_json_token token)
+{
+    int len;
+    char *dupped;
+
+    len = sol_json_token_get_size(&token) -2;
+    dupped = malloc(len); // remove quotes.
+    strncpy(dupped, token.start+1, len);
+    dupped[len] = '\0';
+    return dupped;
+}
+
+static void sol_conffile_set_entry_options(struct sol_conffile_entry *entry, struct sol_json_token options_object)
+{
+    char **options = NULL;
+    int len, i;
+    int numOptions = 0, curr_option = 0;
+    struct sol_json_scanner scanner;
+    struct sol_json_token token, key, value;
+    enum sol_json_loop_reason reason;
+
+    sol_json_scanner_init_from_token(&scanner, &options_object);
+    SOL_JSON_SCANNER_OBJECT_LOOP(&scanner, &token, &key, &value, reason) {
+        numOptions += 1;
+    }
+
+    options = malloc(sizeof(char*) * numOptions);
```

Comment:
because then later I would have to convert from sol_vector to char** again as that's what the options expects.  it's simpler and faster just to iterate once to retrieve the size.